posthog/clickhouse/custom_metrics.py

```diff
@@ -67,3 +71,28 @@ def CUSTOM_METRICS_EVENTS_RECENT_LAG_VIEW():
         AND timestamp < now() + toIntervalMinute(3) AND inserted_at > now() - toIntervalHour(3)
     GROUP BY event;
     """ % {"team_ids": settings.INGESTION_LAG_METRIC_TEAM_IDS}
+
+
+CREATE_CUSTOM_METRICS_COUNTER_EVENTS_TABLE = f"""
+CREATE TABLE IF NOT EXISTS custom_metrics_counter_events (
+    name String,
+    labels Map(String, String),
+    timestamp DateTime64(3, 'UTC') DEFAULT now(),
+    increment Float64
+) ENGINE = {MergeTreeEngine('metrics_counter_events', replication_scheme=ReplicationScheme.REPLICATED)}
+ORDER BY (name, labels, timestamp)
+PARTITION BY toYYYYMM(timestamp)
+"""
+
+CREATE_CUSTOM_METRICS_COUNTERS_VIEW = f"""
+CREATE OR REPLACE VIEW custom_metrics_counters AS
+SELECT
+    name,
+    mapSort(labels) as labels,
+    sum(increment) as value,
+    '' as help,
+    'counter' as type
+FROM custom_metrics_counter_events
+GROUP BY name, type, labels
+ORDER BY name, type, labels
+"""
```

Comment:
We may eventually want to do some pre-aggregation on these if we add more counter types, but for the time being these should be low cardinality enough that aggregation is not a serious issue